forage-android/src/main/java/com/joinforage/forage/android/ui/ForagePANEditText.kt

```diff
@@ -28,23 +28,21 @@ class ForagePANEditText @JvmOverloads constructor(
 ) : AbstractForageElement(context, attrs, defStyleAttr) {
     private val textInputEditText: TextInputEditText
     private val textInputLayout: TextInputLayout
-    private val manager: PanElementStateManager = if (BuildConfig.FLAVOR == "prod") {
-        // strictly support only valid Ebt PAN numbers
-        PanElementStateManager.forEmptyInput()
-    } else {
-        // allows whitelist of special Ebt PAN numbers
-        PanElementStateManager.NON_PROD_forEmptyInput()
-    }
-    private val formatTextWatcher: FormatPanTextWatcher
+
+    private var _SET_ONLY_manager: PanElementStateManager? = null
+    private val manager: PanElementStateManager
+        get() {
+            if (_SET_ONLY_manager == null) {
+                TODO("throw an error here since that means setForageContext was not called")
```

Comment:
Ideally I write a unit test suite that confirms that that `manager` and `_SET_ONLY_manager` work as expected similar to the the unit test suits we have in Forage JS. In the interest of expediency, I was going to gain confidence from qa-tests passing. But lmk what you guys think!

Reply:
I created a ticket for the unit tests [here](https://linear.app/joinforage/issue/FX-589/android-write-unit-tests-for-set-only-field-logic)